### ISSUE TITLE
Add ActivityPub support

### ISSUE THREAD

**Issue by pauloxnet** [wagtail/wagtail#9722]
### Is your proposal related to a problem?

I want the readers of my Wagtail based website will be able to follow my news on Mastodon and other federated platforms that support [ActivityPub](https://activitypub.rocks/).

### Describe the solution you'd like

Add support for ActivityPub in Wagtail so that it works out-of-the-box.

### Describe alternatives you've considered

Wordpress already have a [plugin](https://github.com/pfefferle/wordpress-activitypub) to add ActivityPub support to already published website.

### Additional context

There are already some Django-based implementation of ActivityPub protocol for standalone servers (es: [takahe](https://github.com/jointakahe/takahe) or [bookwyrm](https://github.com/bookwyrm-social/bookwyrm)) but it would be very useful to add ActivtyPub support to an already published Wagtail based website.

**Comment by andrewgodwin** [comment]
So, as an alleged expert in this now, here's what I think the basic coverage of ActivityPub (i.e. sending posts to followers with only a single "account" you follow for the whole server) would entail:

* A Follower model that tracks which ActivityPub Actors are following the server
* A FanOut model that tracks which Actors have been delivered which posts
* A background worker, like Takahē's Stator, that delivers Accept (for Follow) and Create.Note (for posts) messages to other Actors and which can handle retrying them
* An Inbox view that accepts Follow messages from other users, and accepts and drops all the Create.Note messages that other servers are going to start sending you because they think you're more capable than you are.

**Comment by lb-** [comment]
I personally think that something like this would be better suited to be an external package. Maybe even starting as a blog post or two from anyone in the community.

While I agree that this capability would be awesome and I'm all for ActivityPub - this is not something that all sites would use and might not be suitable for Wagtail core. 

It might be worth discussing what parts are missing that may be suitable for Wagtail. For example there is an existing RFC to add background workers to Wagtail - https://github.com/wagtail/rfcs/pull/72

I'll let other core team members add their thoughts and maybe experience with implementing something similar.

**Comment by pauloxnet** [comment]
Thanks for your feedback.
I'm not a Wagtail or ActivityPub expert but I thought that integrating ActivitPub in Wagtail can be a great features that can attract more users and developers.
I opened this issue to start a discussion about this feature but I don't know if it can be implemented as an external package or not, I hope others can share their thoughts.

**Comment by lb-** [comment]
Thanks @pauloxnet - I 100% think this is a good discussion to have and agree that the more Wagtail can do for developers the better it is. However, some things are often best suited as an external package even if it may end up in core in the future. 

Let's give others a week or so to give feedback and thoughts - we may end up moving this to a discussion not an issue from there.

**Comment by pauloxnet** [comment]
I agree with you. Maybe this would be a starting point for an external package, if feasible.
